### PR TITLE
Deleted unnecessary branching

### PR DESCRIPTION
**Pull request by MattiaVerticchio**

### CODE REVIEW

File: pycaret/classification.py [hunk 1]
```diff
@@ -7390,15 +7390,10 @@ def models(
...
-        if type == "linear":
-            df = df[df.index.isin(linear_models)]
-        elif type == "tree":
-            df = df[df.index.isin(tree_models)]
-        elif type == "ensemble":
-            df = df[df.index.isin(ensemble_models)]
+        model_type = {"linear": ["lr", "ridge", "svm"],
+                      "tree": ["dt"],
+                      "ensemble": ["rf", "et", "gbc", "xgboost", "lightgbm", "catboost", "ada"]}
+        df = df[df.index.isin(model_type.get(type))]
```
Comment (Yard1) [inline pycaret/classification.py:7396]: Is this going to work correctly if `type` is None?

**Comment by Yard1** [comment]
Yeah, this doesn't work, @MattiaVerticchio. Check the CI please.

**Comment by MattiaVerticchio** [comment]
Now it should work, I added a default `df.index`, so that returns the input DataFrame when `type` is not in the three specified, including `None`.

**Comment by Yard1** [comment]
Can you make it throw an exception if the type is not one of those three or None?

**Comment by MattiaVerticchio** [comment]
Done

**Comment by Yard1** [comment]
Great, thanks. Gonna wait for CI and merge

File: pycaret/classification.py [hunk 2]
```diff
@@ -7390,16 +7390,16 @@ def models(
...
+        model_type = {"linear": ["lr", "ridge", "svm"],
+                      "tree": ["dt"],
+                      "ensemble": ["rf", "et", "gbc", "xgboost", "lightgbm", "catboost", "ada"]}
+
+        # Check if type is valid
+        if type not in list(model_type):
+            raise ValueError(
+                f"type param only accepts {', '.join(list(model_type))}."
+            )
+    return df[df.index.isin(model_type.get(type, df.index))]
```
Comment (Yard1) [inline pycaret/classification.py:7402]: needs an indent

**Review by Yard1 — CHANGES_REQUESTED** [review]

File: pycaret/classification.py [hunk 3]
```diff
@@ -7390,16 +7390,16 @@ def models(
...
-            df = df[df.index.isin(tree_models)]
-        elif type == "ensemble":
-            df = df[df.index.isin(ensemble_models)]
-        return df
+        model_type = {"linear": ["lr", "ridge", "svm"],
+                      "tree": ["dt"],
+                      "ensemble": ["rf", "et", "gbc", "xgboost", "lightgbm", "catboost", "ada"]}
+
+        # Check if type is valid
+        if type not in list(model_type):
```
Comment (Yard1) [inline pycaret/classification.py:7398]: in list or None, please

**Review by Yard1 — CHANGES_REQUESTED** [review]

File: pycaret/classification.py (hunk 3 above)
Comment (MattiaVerticchio) [reply pycaret/classification.py]: Added `None` as a valid value.

**Comment by Yard1** [comment]
Thanks for working on this, @MattiaVerticchio.

**Pull request merged**